lib/irb/context.rb

```diff
@@ -633,35 +635,60 @@ def evaluate_expression(code, line_no) # :nodoc:
       result
     end
 
-    def parse_command(code)
+    def parse_input(code, is_assignment_expression)
       command_name, arg = code.strip.split(/\s+/, 2)
-      return unless code.lines.size == 1 && command_name
-
       arg ||= ''
-      command = command_name.to_sym
-      # Command aliases are always command. example: $, @
-      if (alias_name = command_aliases[command])
-        return [alias_name, arg]
+
+      # command can only be 1 line
+      if code.lines.size != 1 ||
+        # command name is required
+        command_name.nil? ||
+        # local variable have precedence over command
+        local_variables.include?(command_name.to_sym) ||
+        # assignment expression is not a command
+        (is_assignment_expression ||
+        (arg.start_with?(ASSIGN_OPERATORS_REGEXP) && !arg.start_with?(/==|=~/)))
+        return Statement::Expression.new(code, is_assignment_expression)
       end
 
-      # Assignment-like expression is not a command
-      return if arg.start_with?(ASSIGN_OPERATORS_REGEXP) && !arg.start_with?(/==|=~/)
+      command = command_name.to_sym
 
-      # Local variable have precedence over command
-      return if local_variables.include?(command)
+      # Check command aliases
+      if aliased_name = command_aliases[command]
+        if command_class = Command.load_command(aliased_name)
+          command = aliased_name
+        elsif HelperMethod.helper_methods[aliased_name]
+          message = <<~MESSAGE
+            Using command alias `#{command}` for helper method `#{aliased_name}` is not supported.
+            Please check the value of `IRB.conf[:COMMAND_ALIASES]`.
+          MESSAGE
```

Comment:
How about moving this message generating part into IncorrectAlias and just return `Statement::IncorrectAlias.new(aliased_name)` here

Reply:
But that means we'll be querying registered helper methods from a `Statement` class, which IMO is a bad coupling. I think this type of interaction is better happen inside `Context` or `Irb`?

Reply:
I see. 👍